crates/core/component/dex/src/component/position_manager.rs

```diff
@@ -164,18 +176,13 @@ pub trait PositionManager: StateWrite + PositionRead {
     }
 
     /// Close all positions that have been queued for closure.
-    async fn close_queued_positions(&mut self) -> () {
+    async fn close_queued_positions(&mut self) -> Result<()> {
         let to_close = self.pending_position_closures();
         for id in to_close {
-            match self.close_position_by_id(&id).await {
-                Ok(()) => tracing::debug!(?id, "position closed"),
-                // The position was already closed, which in and of itself is not an error.
-                // It's possible that the position was closed by the engine, for example
-                // because it was a limit-order.
-                Err(e) => tracing::debug!(?id, "failed to close position: {}", e),
-            }
```

Comment:
I believe this logic was incorrect, it would suppress _any_ errors, not just an error about a closed position.

Now `close_position_by_id` enforces its own state transitions internally (allows no-op closures).

Reply:
That match statement was awkward AND !buggy!, good riddance